### ISSUE TITLE
cask-repair: the requested upstream branch does not exists

### ISSUE THREAD

**Issue by MilusW** [vitorgalvao/tiny-scripts#26]
After all steps listed [here](https://github.com/caskroom/homebrew-cask/blob/master/CONTRIBUTING.md#updating-a-cask), cask-repair is not able to create pull request.

Script command:
`cask-repair --pull origin --push MilusW yourkit-java-profiler`

Script output:
```
Submitting…
error: the requested upstream branch 'MilusW' does not exist
hint: 
hint: If you are planning on basing your work on an upstream
hint: branch that already exists at the remote, you may need to
hint: run "git fetch" to retrieve it.
hint: 
hint: If you are planning to push out a new local branch that
hint: will track its remote counterpart, you may want to use
hint: "git push -u" to set the upstream config as you push.
Error creating pull request: Unprocessable Entity (HTTP 422)
Invalid value for "head"
Error creating pull request: Unprocessable Entity (HTTP 422)
Invalid value for "head"
```

git remote -v:
```
MilusW  git@github.com:MilusW/homebrew-cask.git (fetch)
MilusW  git@github.com:MilusW/homebrew-cask.git (push)
origin  https://github.com/caskroom/homebrew-cask (fetch)
origin  https://github.com/caskroom/homebrew-cask (push)
```
brew info cask-repair hub | grep -E '(cask-repair:|hub:)':
```
vitorgalvao/tiny-scripts/cask-repair: stable 0.24.1
hub: stable 2.2.3 (bottled), HEAD
```

I have done everything from #20, but it still doesn't work.

**Comment by vitorgalvao** [comment]
[The branch exists](https://github.com/MilusW/homebrew-cask/branches). Delete it and try again.

**Comment by MilusW** [comment]
I have done it several times - manually and with `cask-repair --push MilusW --delete-branches`. Still the same output.

cask-repair is able to successfully push new branch to my repository, commit changes, but it is unable to create pull request.

**Comment by vitorgalvao** [comment]
Edit your local copy of `cask-repair` and make the line `393` read `git push -u "${remote_push}" "${cask_branch}" --quiet` instead of `git push "${remote_push}" "${cask_branch}" --quiet`.

If that doesn’t work, I’ll have to abandon diagnosing it, as I cannot reproduce. Like in #20, I’ve spent a lot of time on this already, and it was all a waste. What I cannot reproduce I cannot fix unless I am give the cause (which was the case with that issue).

**Comment by MilusW** [comment]
Unfortunately, it doesn't work. Thank you anyway

**Comment by vitorgalvao** [comment]
Thank you for reporting it. If you can get closer to the reason it is happening, I’ll take a look at it, but I’ve pretty much exhausted all the blind suggestions in #20. This is particularly hard to fix since `hub` seems to be the main culprit.

**Comment by jbylsma** [comment]
Sorry if I'm just adding noise by reopening.  I had successfully used `cask-repair` previously, but today was having the same issues trying to update a different formula.

I think I narrowed down the issue to upstream handling in `git branch`, line 394. I exited the script after the failure and did some inspection. I was able to set `origin` as an upstream without a problem, and `master` was automatically used as the tracking branch. However, specifying my `jbylsma` remote would not work, returning

```bash
[/usr/local/Library/Taps/caskroom/homebrew-cask/Casks]
jbylsma$ git branch --set-upstream-to=jbylsma
error: the requested upstream branch 'jbylsma' does not exist
hint:
hint: If you are planning on basing your work on an upstream
hint: branch that already exists at the remote, you may need to
hint: run "git fetch" to retrieve it.
hint:
hint: If you are planning to push out a new local branch that
hint: will track its remote counterpart, you may want to use
hint: "git push -u" to set the upstream config as you push.
```

I reran the git branch using `dtrace` and noticed that there was a failed call to  the remote's HEAD:
```
47746/0x392b1:  lstat64(".git/refs/remotes/jbylsma/HEAD\0", 0x7FFF548E6C70, 0x1000)              = -1 Err#2
```

`origin` still had it's HEAD, which is probably why I could switch to it. I'm not sure when I lost or if I ever had HEAD in `jbylsma`, but maybe that's the root `hub` issue.

Anyways, if the lack of HEAD is the culprit, the easiest fix is to explicitly set to the upstream to `remote/branch`. I was able to successfully use `cask-repair` with the upstream update, and I _think_ the explicit upstream branch shouldn't cause any problems. I'll open a pull request with the change for review.
